### ISSUE TITLE
Feature Request: Custom Loss function with additional arguments

### ISSUE THREAD

**Issue by adamomainz** [catboost/catboost#1601]
Hello,
In order to effectively use the custom loss functionality of Catboost I would like to be able to pass additional parameters into the custom loss function. Is there a way to do the current usage of the catboost custom loss function in python, or can I put in a feature request for this to be a possibility? Similar requests have been made to LightGBM and are accomplished by giving access to the feature-set in the custom loss [in this example](https://github.com/microsoft/LightGBM/issues/4009.). A simple way of doing this might possibly be to pass the index of the rows into the custom loss and allow users to have class level variables at the users discretion.

Operating System: Linux/Docker (and preferably pyspark if possible)
package: Python version 0.24.4
